feature-cancel-insurance/src/main/kotlin/com/hedvig/android/feature/cancelinsurance/CancelInsuranceActivity.kt

```diff
@@ -28,24 +20,19 @@ class CancelInsuranceActivity : AppCompatActivity() {
 
     val insuranceId = intent.getStringExtra(INSURANCE_ID)?.let(::InsuranceId)
       ?: error("Can't open CancelInsuranceActivity without an insurance ID")
-    val viewModel = getViewModel<CancelInsuranceViewModel> { parametersOf(insuranceId) }
 
     setContent {
       HedvigTheme {
-        val uiState by viewModel.uiState.collectAsState()
-        LaunchedEffect(uiState.dateSubmissionSuccess) {
-          if (!uiState.dateSubmissionSuccess) return@LaunchedEffect
-          Toast.makeText(this@CancelInsuranceActivity, "Navigate to success screen", Toast.LENGTH_LONG).show()
-        }
-        CancelInsuranceScreen(
+        CancelInsuranceNavHost(
           windowSizeClass = calculateWindowSizeClass(this),
-          datePickerState = uiState.datePickerState,
-          dateValidator = viewModel.dateValidator,
-          canSubmit = uiState.canContinue,
-          submit = viewModel::submitSelectedDate,
-          hasError = uiState.dateSubmissionError,
-          showedError = viewModel::showedError,
-          navigateBack = { onBackPressedDispatcher.onBackPressed() },
+          navController = rememberNavController(),
+          insuranceId = insuranceId,
+          navigateUp = {
+            if (!onSupportNavigateUp()) {
+              finish()
+            }
+            true
+          },
```

Comment:
Context here https://developer.android.com/guide/navigation/navigation-navigate#back-stack
Doing navigateUp() is what's expected by the system when clicking on the topAppBar back icon, as that should *not* exit the app if thats at all possible, but if `onSupportNavigateUp` returns false it means that that was not possible, so then we fallback to finish(). Taken from the docs linked above + this discussion https://slack-chats.kotlinlang.org/t/9661532/is-there-a-way-when-using-navigation-compose-to-immitate-exa#a951c61b-a697-4cd3-9f03-6299d916ec6d